Sources/MapLibreSwiftUI/MapViewCoordinator.swift

```diff
@@ -188,25 +195,35 @@ extension MapViewCoordinator: MLNMapViewDelegate {
 
     /// The MapView's region has changed with a specific reason.
     public func mapView(_ mapView: MLNMapView, regionDidChangeWith reason: MLNCameraChangeReason, animated _: Bool) {
-        // Validate that the mapView.userTrackingMode still matches our desired camera state for each tracking type.
-        let isFollowing = parent.camera.state == .trackingUserLocation && mapView.userTrackingMode == .follow
-        let isFollowingHeading = parent.camera.state == .trackingUserLocationWithHeading && mapView
-            .userTrackingMode == .followWithHeading
-        let isFollowingCourse = parent.camera.state == .trackingUserLocationWithCourse && mapView
-            .userTrackingMode == .followWithCourse
-
         // If any of these are a mismatch, we know the camera is no longer following a desired method, so we should
-        // detach and revert to a .centered camera. If any one of these is true, the desired camera state still matches
-        // the mapView's userTrackingMode
-        if isFollowing || isFollowingHeading || isFollowingCourse {
-            // User tracking is still active, we can ignore camera updates until we unset/fail this boolean check
+        // detach and revert to a .centered camera. If any one of these is true, the desired camera state still
+        // matches the mapView's userTrackingMode
+        let isProgrammaticallyTracking: Bool = switch parent.camera.state {
+        case .centered(onCoordinate: _):
+            false
+        case .trackingUserLocation:
+            mapView.userTrackingMode == .follow
```

Comment:
I'm going to improve the comments on what happens here in the next commit. It's changed a bit since the review, but the point of this method is to update the two-way binding to _reflect the actual camera state_. This set of checks is to preserve the intent ("following") when the parent state (original) agrees with the map view.

What follows is to update the parent camera to a mode that just describes the state of the map (centered on a point with certain viewport characteristics). It's just telling the parent binding "this is how things actually are" but doesn't result in a UI change.